galsim/compound.py

```diff
@@ -645,6 +744,33 @@ def __str__(self):
         s += ')'
         return s
 
+    def _prepareDraw(self):
+        self._orig_obj._prepareDraw()
+        self.SBProfile = galsim._galsim.SBAutoConvolve(self._orig_obj.SBProfile, self._real_space,
+                                                       self._gsparams)
+
+    def shoot(self, n_photons, rng=None):
+        """Shoot photons into a PhotonArray.
+
+        @param n_photons    The number of photons to use for photon shooting.
+        @param rng          If provided, a random number generator to use for photon shooting,
+                            which may be any kind of BaseDeviate object.  If `rng` is None, one
+                            will be automatically created, using the time as a seed.
+                            [default: None]
+        @returns PhotonArray.
+        """
+        # Setup the rng if not provided one.
+        if rng is None:
```

Comment:
Okay, I found (and [committed](https://github.com/GalSim-developers/GalSim/pull/847/commits/c5629f99130b06bcfd170eafe19e3dc5fa6fa3af)) what I think is a better solution: add a `.def("__init__" ...)` using `make_constructor` as the lowest priority (so first in order) initializer that captures any python object as the seed, check if it's `None`, and throw if not.  This solves the virtual override problem for `DistDeviate`.

I'm getting a new (old) error though now in `test_config_image.py:test_njobs()` similar(identical?) to that found [here](https://github.com/GalSim-developers/GalSim/pull/714#issuecomment-190443809).  Basically, you get different results if you split up a config job into separate processes than if you run them through the same process.  Looking at the debug output, it seems like the random seeds are consistent, but as before, the number of cosmos gals passing cuts is not consistent.  Thoughts?